### DISCUSSION TITLE
Issue with TransferListener Callbacks Not Triggering for CopyObjectRequest

### DISCUSSION THREAD

**Discussion by supremvanam** [aws/aws-sdk-java-v2#5124] [Q&A]
Hi,

I am currently working with the AWS SDK for Java and leveraging the TransferListener interface to monitor S3 operations. I've successfully implemented and tested the TransferListener's callbacks during Upload and Download operations where all four methods (transferInitiated, bytesTransferred, transferComplete, and transferFailed) are appropriately called. However, I'm facing a peculiar issue with the CopyObjectRequest.

When performing a CopyObjectRequest, only the transferInitiated and transferFailed callback is triggered, and the other two (bytesTransferred, transferComplete) are not triggered, even though the copy operation completes successfully (I have verified the files in both source and destination locations). This behaviour is consistent regardless of the object size or the region.

Could anyone explain why CopyObjectRequest might behave differently in triggering TransferListener callbacks? Is there a different approach or a specific configuration required to fully utilize TransferListener for copy operations?

Thank you in advance for your insights and suggestions!

```
List<CompletableFuture<CompletedCopy>> copyFutures = new ArrayList<>();

List<CopyRequest> requests = sourceObjects.stream().map(object ->
        CopyObjectRequest.builder()
                .sourceBucket(bucketName())
                .sourceKey(object.sourceKey())
                .destinationBucket(bucketName()) // same bucket, different key
                .destinationKey(object.destinationKey())
                .ssekmsKeyId(keyId())
                .serverSideEncryption(serverSideEncryption)
                .storageClass(storageClassAsString())
                .metadataDirective(MetadataDirective.REPLACE)
                .bucketKeyEnabled(true)
                .build()
).map(object -> CopyRequest.builder()
        .addTransferListener(LoggingTransferListener.create())
        .copyObjectRequest(object)
        .build()).toList();

copyFutures.addAll(requests
                    .stream()
                    .map(transferManager::copy)
                    .map(Copy::completionFuture)
                    .toList());
```

**Comment by saisaran96** [comment]
Hi,
I am also facing the same issue. Since S3-to-S3 CopyRequest uses a server-side copy, the TransferListener methods are not getting called.
Just wanted to check, were you able to solve this issue? Do you have any workaround for this?
